torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -849,6 +849,127 @@ bool TensorDomain::hasReduction(const std::vector<IterDomain*>& td) {
   return false;
 }
 
+// return mapping of consumer_domain[i] = producer_domain[result_vector[i]]
+// assuming there exists a direct consumer-producer mapping. If axis exists in
+// consumer (broadcast) but not in producer, mapping will be result_vector[i] =
+// -1.
+std::vector<int64_t> TensorDomain::mapDomainCtoP(
```

Comment:
This seems very similar to `mapDomainPtoC`. I have some refactoring idea. Will send a PR once this is merged.